subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationQueue.java

```diff
@@ -129,20 +148,52 @@ private static String getFailureMessage(Collection<? extends Throwable> failures
         return "Multiple build operations failed.";
     }
 
+    private class SubmittedWorker {
```

Comment:
Can you clarify what you're suggesting here?  The problem we are trying to solve with this is that a worker may not actually make it onto the executor thread pool.  So when waiting, we know that the operation queue is empty, but we don't know if operations are still running in workers.  We need to know whether a worker has started (and we need to wait for it to finish) or if it has never started and is stuck in the executor queue.  We don't want to wait on workers stuck in the queue because we don't know when they will actually execute and we know they won't process any work anyways.  For that reason, I don't think we can just use a workers counter - I think we need to be able to query individual workers.